src/components/forms/DateField/DatePicker/DatePicker.tsx

```diff
@@ -139,17 +138,25 @@ const DatePicker: React.FC<DatePickerProps> = ({
       <Paragraph className="openforms-datepicker-textbox">
         <Textbox
           name={name}
-          value={textboxValue}
-          onChange={onChange}
+          value={rawValue && parseDate(rawValue) ? _formatDate(rawValue) : rawValue}
+          onChange={event => {
+            // Set the raw value in the textbox and update the formik value depending on
+            // if we have a valid date or not
+            const newRaw = event.target.value;
+            setRawValue(newRaw);
+            const parsed = parseDate(newRaw, dateLocaleMeta);
+            setValue(parsed ? formatISO(parsed, {representation: 'date'}) : newRaw);
+          }}
           onBlur={async event => {
-            const value = event.target.value;
             // Attempt to create a date object using the locale meta
-            const date = parseDate(value, dateLocaleMeta);
-            // If we were able to create a date object, format it to an ISO-8601 string and set it
-            // as the field value. Otherwise, just set the entered value to the field directly.
-            // It's up to the validation libraries to check it.
-            const newValue = date ? formatISO(date, {representation: 'date'}) : value;
-            await setValue(newValue);
+            const parsed = parseDate(rawValue, dateLocaleMeta);
+            if (parsed) {
+              // We have a date object, format it according to the locale (and remove
+              // comma) and use as the textbox value.
```

Comment:
```suggestion
              // We have a date object, format it according to the locale and use as the textbox value.
```
Probably copied over from the datetime component?